packages/rum/src/boot/postStartStrategy.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  SessionReplayState,
+  type LifeCycle,
+  type RumConfiguration,
+  type RumSessionManager,
+  type StartRecordingOptions,
+  type ViewHistory,
+} from '@datadog/browser-rum-core'
+import { runOnReadyState, type DeflateEncoder } from '@datadog/browser-core'
+import { getSessionReplayLink } from '../domain/getSessionReplayLink'
+import type { startRecording } from './startRecording'
+
+export type StartRecording = typeof startRecording
+
+export const enum RecorderStatus {
+  // The recorder is stopped.
+  Stopped,
+  // The user started the recording while it wasn't possible yet. The recorder should start as soon
+  // as possible.
+  IntentToStart,
+  // The recorder is starting. It does not record anything yet.
+  Starting,
+  // The recorder is started, it records the session.
+  Started,
+}
+export type RecorderState = {
+  status: RecorderStatus
+  setStatus: (status: RecorderStatus) => void
+  statusEqual: (status: RecorderStatus) => boolean
+}
+
+export interface Strategy {
+  startStrategy: (options?: StartRecordingOptions) => void
+  stopStrategy: () => void
+  getSessionReplayLinkStrategy: () => string | undefined
+}
+
+export function createPostStartStrategy(
+  configuration: RumConfiguration,
+  lifeCycle: LifeCycle,
+  sessionManager: RumSessionManager,
+  viewHistory: ViewHistory,
+  startRecordingImpl: StartRecording,
+  getOrCreateDeflateEncoder: () => DeflateEncoder | undefined,
+  state: RecorderState
+): Strategy {
+  let stopRecording: () => void
+  return {
+    startStrategy(options?: StartRecordingOptions) {
+      const session = sessionManager.findTrackedSession()
+      if (!session || (session.sessionReplay === SessionReplayState.OFF && (!options || !options.force))) {
+        state.setStatus(RecorderStatus.IntentToStart)
+        return
+      }
+
+      if (state.statusEqual(RecorderStatus.Starting) || state.statusEqual(RecorderStatus.Started)) {
+        return
+      }
+
+      state.setStatus(RecorderStatus.Starting)
+
+      runOnReadyState(configuration, 'interactive', () => {
+        if (state.status !== RecorderStatus.Starting) {
+          return
+        }
+
+        const deflateEncoder = getOrCreateDeflateEncoder()
+        if (!deflateEncoder) {
+          state.setStatus(RecorderStatus.Stopped)
+          return
+        }
+
+        ;({ stop: stopRecording } = startRecordingImpl(
+          lifeCycle,
+          configuration,
+          sessionManager,
+          viewHistory,
+          deflateEncoder
+        ))
+
+        state.setStatus(RecorderStatus.Started)
+      })
+
+      if (options && options.force && session.sessionReplay === SessionReplayState.OFF) {
+        sessionManager.setForcedReplay()
+      }
+    },
+
+    stopStrategy() {
+      if (state.statusEqual(RecorderStatus.Stopped)) {
+        return
+      }
+
+      stopRecording?.()
```

Comment:
The original version of this code did this:
```ts
        if (state.status === RecorderStatus.Stopped) {
          return
        }
        if (state.status === RecorderStatus.Started) {
          state.stopRecording()
        }
```

Since `RecorderState` was a discriminated union, it was necessary to check that `state.status` was exactly `RecorderStatus.Started` before calling `stopRecording()`, so we were guaranteed to only ever call `stopRecording()` in that state.

Now, we check that `stopRecording` is truthy, but we don't verify that we're in state `RecorderStatus.Started`. We set `stopRecording` when we _enter_ `RecorderStatus.Started`, but we don't clear it when we _exit_ `RecorderStatus.Started`, so checking if it's truthy just tells us that we've been in `RecorderStatus.Started` at some point in the past.

It's possible that the overall flow of the code means that this isn't an issue, but from a belt-and-suspenders perspective it feels like it might be best to reintroduce some sort of check here that we're in the expected state. Perhaps something like this?
```suggestion
      if (state.statusEqual(RecorderStatus.Started)) {
        const stopRecordingNow = stopRecording
        stopRecording = undefined
        stopRecordingNow?.()
      }
```

This way we'd both ensure that we're in the expected state, and clear `stopRecording` so that it's not accidentally reused later or invoked in a reentrant manner. (Similar guarantees to what we got from the discriminated union before, in other words.)

Reply:
If I'm reading that correctly, it boils down to:
```ts
if (status === RecorderStatus.Started) {
  stopRecording?.()
}

status = RecorderStatus.Stopped
```

I think that's OK; it's quite similar to the old behavior! We do leave the old `stopRecording` value around, but that's probably OK, since if we reenter `RecorderStatus.Started`, we'll overwrite it.

Reply:
Oh my 🤦‍♂️. Yes, let's go with 
```ts
if (status === RecorderStatus.Started) {
  stopRecording?.()
}

status = RecorderStatus.Stopped
```

Reply:
SGTM!